theano/sandbox/gpuarray/dnn.py

```diff
@@ -1038,69 +1044,89 @@ def c_code_cache_version(self):
 
 
 class GpuDnnPool(DnnBase):
-    """
-    Pooling.
 
+    """
     Parameters
     ----------
     img
-        The image 4d tensor.
-    desc
-        The pooling descriptor.
-
+        The image 4d or 5d tensor.
+    Parameters
+    ----------
+    ws : tensor variable
+        Window size.
+    stride : tensor variable
+        (dx, dy) or (dx, dy, dz).
+    mode : {'max', 'average_inc_pad', 'average_exc_pad'}
+        The old deprecated name 'average' corresponds to 'average_inc_pad'.
+    pad : tensor
+        (padX, padY) or (padX, padY, padZ)
     """
 
-    __props__ = ()
+    __props__ = ('mode',)
 
-    def __init__(self):
+    def __init__(self, mode='max'):
         DnnBase.__init__(self, ["dnn_pool.c"], "APPLY_SPECIFIC(dnn_pool)")
+        if mode == 'average':
+            mode = 'average_inc_pad'
+        assert mode in ('max', 'average_inc_pad', 'average_exc_pad')
+        if version() == -1:
+            raise Exception("cudnn v1 do not support average_exc_pad")
```

Comment:
remove this check, we don't support cudnn v1 anymore. The last supported cudnn is v3, if you find other stuff for other cudnn version, you can remove them too.